core/common-ui/src/main/java/team/noweekend/core/common/ui/calendar/util/CalendarUtils.kt

```diff
@@ -0,0 +1,70 @@
+package team.noweekend.core.common.ui.calendar.util
+
+import kotlinx.datetime.Clock
+import kotlinx.datetime.DateTimeUnit
+import kotlinx.datetime.DayOfWeek
+import kotlinx.datetime.LocalDate
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.minus
+import kotlinx.datetime.number
+import kotlinx.datetime.plus
+import kotlinx.datetime.toLocalDateTime
+
+object CalendarUtils {
+
+    fun now(): LocalDate {
+        return Clock.System.now().toLocalDateTime(TimeZone.currentSystemDefault()).date
+    }
+
+    fun LocalDate.minusWeeks(week: Int): LocalDate {
+        return this.minus(week, DateTimeUnit.WEEK)
+    }
+
+    fun LocalDate.plusWeeks(week: Int): LocalDate {
+        return this.plus(week, DateTimeUnit.WEEK)
+    }
+
+    fun LocalDate.minusMonths(month: Int): LocalDate {
+        return this.minus(month, DateTimeUnit.MONTH)
+    }
+
+    fun LocalDate.plusMonths(month: Int): LocalDate {
+        return this.plus(month, DateTimeUnit.MONTH)
+    }
+
+    fun LocalDate.plusDays(days: Int): LocalDate {
+        return this.plus(days, DateTimeUnit.DAY)
+    }
+
+    fun LocalDate.firstDayOfMonth(): LocalDate {
+        return LocalDate(this.year, this.month, 1)
+    }
+
+    fun LocalDate.lastDayOfMonth(): LocalDate {
+        val lastDay = this.month.number.monthLength(isLeapYear(year))
+        return LocalDate(this.year, this.month, lastDay)
+    }
+
+    fun LocalDate.previousOrSame(dayOfWeek: DayOfWeek): LocalDate {
+        val daysDiff = (this.dayOfWeek.value - dayOfWeek.value + 7) % 7
+        return this.minus(daysDiff.toLong(), DateTimeUnit.DAY)
+    }
+
+    fun LocalDate.nextOrSame(dayOfWeek: DayOfWeek): LocalDate {
+        val daysDiff = (dayOfWeek.value - this.dayOfWeek.value + 7) % 7
+        return this.plus(daysDiff.toLong(), DateTimeUnit.DAY)
+    }
+
+    private fun Int.monthLength(isLeapYear: Boolean): Int {
+        return when (this) {
+            2 -> if (isLeapYear) 29 else 28
+            4, 6, 9, 11 -> 30
+            else -> 31
+        }
+    }
```

Comment:
```suggestion
    private fun Int.monthLength(isLeapYear: Boolean): Int {
        return Month.of(this).length(isLeapYear)
    }
```

작성해준 코드 보고 내가 이해하기로는 monthLength 수신객체 Int가 특정 월의 숫자를 의미하는 걸로 이해했는데, monthLength를 윤년 기준으로 구한다고 하면 매직넘버를 제거하고 위처럼 작성할 수 있을 것 같은데 어땡

Reply:
@JaesungLeee 훨낫다 반영해볼께

Reply:
@JaesungLeee 
[7ba367e](https://github.com/mash-up-kr/NoWeekend-Android/pull/32/commits/7ba367e228d79df338f5679ae83d3307d0280d5c)
반완